### DISCUSSION TITLE
Information of why changepoint was detected

### DISCUSSION THREAD

**Discussion by ctselas7** [deepcharles/ruptures#253] [Q&A] [answered]
I am not an expert on the algorithms for changepoint detection. But do you think it's possible that we can get the reason why a changepoint was detected or how certain the algorithm is? I can understand that for a more complicated algorithm, one simple answer can not be easily given. But has anyone an idea for this?

**Comment by deepcharles** [comment, marked as answer]
Detected changes depend on the cost function that you use. CostL2 will find changes in the mean, CostNormal will find changes in the mean and variance/covariance, etc. For each cost function, the type of change that can be detected is described in [the documentation](https://centre-borelli.github.io/ruptures-docs/user-guide/).

For a given cost function, the chosen indexes are chosen because they minimize a criterion (called the sum of costs, see [this review article](https://charles.doffy.net/publication/sp-review-2020) associated with this Python library).

**Comment by ctselas7** [comment]
@deepcharles thanks! Now everything is more clear.
